pycozmo/activity.py

```diff
@@ -14,30 +16,389 @@
 ]
 
 
+class BehaviorChooser:
+
+    def __init__(self,
+                 choice_type: str,
+                 behaviors: List) -> None:
+
+        self.choice_type = choice_type
+        self.behaviors = behaviors
+        self.iteration = 0
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(
+            choice_type=data['type'],
+            behaviors=data['behaviors'] if 'behaviors' in data else []
+        )
+
+    def reset(self):
+        self.iteration = 0
+
+    def choose(self):
+        if self.choice_type == 'Selection':
+            return None
+        if self.choice_type == 'StrictPriority':
+            if self.iteration <= len(self.behaviors):
+                out = self.behaviors[self.iteration]
+                self.iteration += 1
+                return out
+            else:
+                return None
+        elif self.choice_type == 'Scoring':
+            # TODO: add score based choice method
+            pass
+        else:
+            print(self.choice_type)
+        return None
+
+
+class Objective:
+    def __init__(self,
+                 objective: str,
+                 behavior_ID: str,
+                 ignore_if_locked: str,
+                 probability_to_require_objective: float,
+                 random_completions_needed_min: Optional[int] = 0,
+                 random_completions_needed_max: Optional[int] = 0) -> None:
+        self.objective = objective
+        self.behavior_ID = behavior_ID
+        self.ignore_if_locked = ignore_if_locked
+        self.probability_to_require_objective = probability_to_require_objective
+        self.random_completions_needed_min = random_completions_needed_min
+        self.random_completions_needed_max = random_completions_needed_max
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(objective=data['objective'],
+                   behavior_ID=data['behaviorID'],
+                   ignore_if_locked=data['ignoreIfLocked'],
+                   probability_to_require_objective=data['probabilityToRequireObjective'],
+                   random_completions_needed_min=data['randomCompletionsNeededMin']
+                   if 'randomCompletionsNeededMin' in data else None,
```

Comment:
If the default is 0, why pass None here?

Reply:
This is the same as `random_completions_needed_min=data.get('randomCompletionsNeededMin')

Reply:
It's None due to copy-paste or autocomplete most likely. I'll correct it.